src/Shared/RoslynUtils/TypeHelper.cs

```diff
@@ -36,4 +37,33 @@ internal static bool IsCompilerGeneratedMethod(MethodInfo method)
     {
         return Attribute.IsDefined(method, typeof(CompilerGeneratedAttribute)) || IsCompilerGeneratedType(method.DeclaringType);
     }
+
+    /// <summary>
+    /// Tries to get non-compiler-generaged name of function. This parses generated local function names out of a generated method name if possible.
+    /// </summary>
+    internal static bool TryGetNonCompilerGeneratedMethodName(MethodInfo method, [NotNullWhen(true)] out string? originalName)
```

Comment:
This is shared source, but I think this is the only caller so I can make it private. I like calling the method over doing `Attribute.IsDefined(...) || IsCompilerGeneratedType(...)` inline, because I think it's a little more self-documenting.